### ISSUE TITLE
How is passed value from --connection argument from migrations bundle into "bundled" migration project?

### ISSUE THREAD

**Issue by petrciml** [dotnet/efcore#28692]
## Description:
I have a DB migrations specific project (not shared with app code).

Program.cs is quite easy, (trying to use Options pattern):
```C#
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

var builder = Host.CreateDefaultBuilder(args);
builder.ConfigureServices((hostContext, services) =>
{
    services.AddSingleton<MigrationContext>();
    services.Configure<MigrationOptions>(hostContext.Configuration)
        .PostConfigure<MigrationOptions>(mo =>
            mo.Connection = hostContext.Configuration.GetConnectionString("MyDb"));
});
var app = builder.Build();
await app.RunAsync();
```

MigrationOptions.cs:
```C#
public class MigrationOptions
{
    public string Connection { get; set; } = null!;
    public bool Debug { get; set; }
}
```
MigrationContext.cs (ctor only):
```C#
public MigrationContext(IOptions<MigrationOptions> option) : base(
        new DbContextOptionsBuilder<MyDbContext>()
            .EnableSensitiveDataLogging()
            .EnableDetailedErrors()
            .UseLoggerFactory(logFactory)
            .UseSqlServer(**option.Value.Connection**, o => 
                o.CommandTimeout((int)TimeSpan.FromMinutes(3).TotalSeconds))
            .Options)
    {
        if (option.Value.Debug)
        {
            Console.Out.WriteLine("Current ProcessID: " + Environment.ProcessId); //This prints the process id
            Console.Out.WriteLine("Waiting for debugger to attach...");
            while (!Debugger.IsAttached)
            {
                Thread.Sleep(1000);
            }
            Console.Out.WriteLine("Debugger attached!");
        }
    }
```

With appsetting.json file in the folder everything works great. 
But in case of bundled without appsetting.json only trying specify connection via --connection argument the error occurred:

```
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Finding application service provider in assembly 'Cardif.MessageBus.DbStructure'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Finding DbContext classes in the project...
Found DbContext 'MigrationContext'.
Microsoft.EntityFrameworkCore.Design.OperationException: Unable to create an object of type 'MigrationContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'connectionString')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer(DbContextOptionsBuilder optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Microsoft.EntityFrameworkCore.SqlServerDbContextOptionsExtensions.UseSqlServer[TContext](DbContextOptionsBuilder`1 optionsBuilder, String connectionString, Action`1 sqlServerOptionsAction)
   at Cardif.MessageBus.DbStructure.MigrationContext..ctor(IOptions`1 option)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetServiceOrCreateInstance(IServiceProvider provider, Type type)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.<>c__DisplayClass21_4.<FindContextTypes>b__13()
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.<>c__DisplayClass21_4.<FindContextTypes>b__13()
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String connectionString, String contextType)
   at Microsoft.EntityFrameworkCore.Migrations.Design.MigrationsBundle.ExecuteInternal(String[] args)
   at Microsoft.EntityFrameworkCore.Migrations.Design.MigrationsBundle.<>c__DisplayClass6_0.<Configure>b__0(String[] args)
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.EntityFrameworkCore.Migrations.Design.MigrationsBundle.Execute(String context, Assembly assembly, Assembly startupAssembly, String[] args)
**Unable to create an object of type 'MigrationContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728**
```

Please how to change the code of migration project to access value of connection string from --connection argument given to the bundle? If it's possible via Options pattern

**Comment by ajcvickers** [comment]
Duplicate of #26869.

**Comment by petrciml** [comment]
> Duplicate of #26869.
> 
> Workarounds before EF7:
> 
> * Deploy  appsetting.json with the bundle. This is recommended anyway.
> * Add conditional code to not call `UseSqlServer` with null.

Yah, very great answer, thanks a lot...
It seems that everything works well now!
My solution (for the others interest):

```C#
    // Better solution will be usage of IDesignTimeDbContextFactory<MigrationContext> (as you can see in git history)
    //  but the way through the factory isn't compatible with dependency injection and we need the DI for Options pattern.
    // Why we need Options pattern? For the ensuring same behavior in bundle when appsettings.json is used, env. vars used or cmd line args used.
    public MigrationContext(IOptions<MigrationOptions> option) : base(ConfigureContext(option))
    {
        if (option.Value.Debug)
        {
            Console.Out.WriteLine("Current ProcessID: " + Environment.ProcessId); //This prints the process id
            Console.Out.WriteLine("Waiting for debugger to attach...");
            while (!Debugger.IsAttached)
            {
                Thread.Sleep(1000);
            }
            Console.Out.WriteLine("Debugger attached!");
        }
    }

    private static DbContextOptions<MessageBusContext> ConfigureContext(IOptions<MigrationOptions> migrationAppOptions)
    {
        var preconfiguredBuilder = new DbContextOptionsBuilder<MyContext>()
            .EnableSensitiveDataLogging()
            .EnableDetailedErrors()
            .UseLoggerFactory(logFactory);

        void SqlServerOptionsAction(SqlServerDbContextOptionsBuilder o) => o.CommandTimeout((int) TimeSpan.FromMinutes(3).TotalSeconds);

        //when going through migrations bundle the connection string coming from "--connection" argument
        // will be changed later on already existing context instance (in OnConfiguration event)
        //see following question & related issue (will be fixed in EF Core 7.0):
        // https://github.com/dotnet/efcore/issues/28692
        // https://github.com/dotnet/efcore/issues/26869
        return string.IsNullOrEmpty(migrationAppOptions.Value.Connection)
            ? preconfiguredBuilder.UseSqlServer(SqlServerOptionsAction).Options
            : preconfiguredBuilder.UseSqlServer(migrationAppOptions.Value.Connection, SqlServerOptionsAction).Options;
    }
```
